src/NuGet.Core/NuGet.Build.Tasks/GetRestoreProjectStyleTask.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Build.Framework;
+using NuGet.Commands;
+using NuGet.ProjectModel;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace NuGet.Build.Tasks
+{
+    /// <summary>
+    /// Gets the project style.
+    /// </summary>
+    public sealed class GetRestoreProjectStyleTask : Task
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the project has any PackageReference items.
+        /// </summary>
+        public bool HasPackageReferenceItems { get; set; }
+
+        /// <summary>
+        /// Gets or sets the full path to the project directory.
+        /// </summary>
+        [Required]
+        public string MSBuildProjectDirectory { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the project file.
+        /// </summary>
+        [Required]
+        public string MSBuildProjectName { get; set; }
+
+        [Output]
+        public bool PackageReferenceCompatibleProjectStyle { get; set; }
+
+        /// <summary>
+        /// The path to a project.json file.
+        /// </summary>
+        public string ProjectJsonPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="NuGet.ProjectModel.ProjectStyle"/> of the project.
+        /// </summary>
+        [Output]
+        public ProjectStyle ProjectStyle { get; set; }
+
+        /// <summary>
+        /// Gets or sets the user specified project style of the project.
+        /// </summary>
+        public string RestoreProjectStyle { get; set; }
+
+        public override bool Execute()
+        {
+            var log = new MSBuildLogger(Log);
+
+            // Log Inputs
+            BuildTasksUtility.LogInputParam(log, nameof(HasPackageReferenceItems), HasPackageReferenceItems.ToString());
+            BuildTasksUtility.LogInputParam(log, nameof(MSBuildProjectDirectory), MSBuildProjectDirectory);
+            BuildTasksUtility.LogInputParam(log, nameof(MSBuildProjectName), MSBuildProjectName);
+            BuildTasksUtility.LogInputParam(log, nameof(ProjectJsonPath), ProjectJsonPath);
+            BuildTasksUtility.LogInputParam(log, nameof(RestoreProjectStyle), RestoreProjectStyle);
```

Comment:
Why not do something like [this](https://github.com/NuGet/NuGet.Client/pull/3111/files#diff-db1b469e63dbc5a9d027fca99cb23294R31)?
```C#
Log.LogMessageFromResources(HasPackageReferenceItems, MSBuildProjectDirectory, MSBuildProjectName, ProjectJsonPath, RestoreProjectStyle);
```

I'm unfamiliar with both approaches, so it's mainly a learning opportunity for me.

Reply:
`Log.LogMessageFromResources` is a helper in MSBuild to look up strings and log them.  `BuildTasksUtility.LogInputParam` calls into NuGet's `ILogger` which has an `MSBuildLogger` implementation.  

In this particular case, I copied/pasted code from other tasks to keep things similar, this is how they all log input parameters.